Count/ViewController.swift

```diff
@@ -10,47 +10,65 @@ import UIKit
 
 class ViewController: UIViewController {
     
-    var number: Int = 0
     @IBOutlet var label: UILabel!
-
+    @IBOutlet var bestView: BestLabel!
+    private var timer: Timer?
+    private var timerTotalDuration: TimeInterval = 0
+    var db = UserDefaults.standard
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        db.register(defaults: ["highestScore":0.0])
+        bestView.text = String(format: "%.2f",db.double(forKey: "highestScore"))
     }
-
+    
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
     
-    @IBAction func plus() {
-        
-        number = number + 3
-        label.text = String(number)
+    @IBAction func startBtn(){
+        startTimer()
     }
     
-    /* Check.1 マイナス機能 */
-    @IBAction func minus() {
-        
-        number = number - 1
-        label.text = String(number)
+    @IBAction func stopBtn(){
+        stopTimer()
     }
     
-    /* Check.2 かける・割る機能 */
-    @IBAction func times() {
-        
-        number = number * 2
-        label.text = String(number)
+    @IBAction func resetBtn(){
+        if db.double(forKey: "highestScore") < timerTotalDuration {
+            db.set(timerTotalDuration,forKey:"highestScore")
+        }
+        resetTimer()
+        label.text = "0.00"
+        bestView.text = String(format: "%.2f",db.double(forKey: "highestScore"))
     }
-
-    @IBAction func divide() {
-        
-        number = number / 2
-        label.text = String(number)
+    
+    func startTimer() {
+        guard timer == nil else { return }
```

Comment:
```suggestion
        guard let timer = timer else { return }
```

か

```suggestion
        if timer != nil else { return }
```

のほうがよく使われると思うので早期リターンするならこっちかなぁ